src/main/java/de/tum/in/www1/artemis/versioning/VersionRangesRequestCondition.java

```diff
@@ -0,0 +1,342 @@
+        // As long as there are ranges in the pool, pop the first one and check against all other ranges
+        // If there is an overlap, remove the second range, combine the two ranges, and add the new range to the pool
+        while (!rangePool.isEmpty()) {
+            var selectedRange = rangePool.remove(0);
+            boolean combined = false;
+            for (VersionRange range : rangePool) {
+                boolean canCombine = switch (VersionRangeComparator.compare(selectedRange, range)) {
+                    case FIRST_A_NO_INTERSECT, FIRST_B_NO_INTERSECT -> false;
+                    default -> true;
+                };
+                if (canCombine) {
+                    rangePool.remove(range);
+                    rangePool.add(VersionRangeFactory.combine(selectedRange, range));
+                    combined = true;
+                    break;
+                }
+            }
+            if (!combined) {
+                newRanges.add(selectedRange);
+            }
+        }
+        return newRanges;
+    }
+
+    /**
+     * Checks if the given condition collides with this condition.
+     *
+     * @param versionRangesRequestCondition The condition to check.
+     * @return True if the conditions collide, false otherwise.
+     */
+    public boolean collide(@NotNull VersionRangesRequestCondition versionRangesRequestCondition) {
+        Set<VersionRange> tbcRanges = versionRangesRequestCondition.getRanges();
+
+        // If one of those represents all versions, the other one has to collide
+        if (ranges.isEmpty() || tbcRanges.isEmpty()) {
+            return true;
+        }
+
+        for (VersionRange range : ranges) {
+            for (VersionRange tbcRange : tbcRanges) {
+                if (inRangeCodes.contains(VersionRangeComparator.compare(range, tbcRange))) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    public Set<VersionRange> getRanges() {
+        return ranges;
+    }
+
+    @Override
+    public int compareTo(@NotNull VersionRangesRequestCondition versionRangesRequestCondition, @NotNull HttpServletRequest httpServletRequest) {
+        if (this.getMatchingCondition(httpServletRequest) == this) {
+            return -1;
+        }
+        else if (versionRangesRequestCondition.getMatchingCondition(httpServletRequest) == versionRangesRequestCondition) {
+            return 1;
+        }
+        else {
+            return 0;
+        }
+    }
+
+    /**
+     * Returns the condition that matches the request or {@code null} if there is no match.
+     *
+     * @param request the current request
+     * @return the matching condition or {@code null}
+     */
+    @Nullable
+    @Override
+    public VersionRangesRequestCondition getMatchingCondition(@NotNull HttpServletRequest request) {
+        String path = request.getRequestURI();
+        if (!path.matches("^/?api/.*")) {
```

Comment:
```suggestion
        if (!path.startsWith("api/") && !path.startsWith("/api/")) {
```
The regex can be avoided easily here.